src/client/eppo-client.ts

```diff
@@ -154,6 +163,28 @@ export default class EppoClient {
     this.isObfuscated = isObfuscated;
   }
 
+  async withOverrides(overridePayload: string | undefined): Promise<EppoClient> {
+    if (overridePayload) {
+      try {
+        const { apiKey, overrides } = JSON.parse(overridePayload) as OverridePayload;
+        const isValidApiKey = await this.overrideValidator.validateOverrideApiKey(apiKey);
+        if (!isValidApiKey) {
+          // TODO - handle error
+          return this;
+        }
+        if (overrides && Object.keys(overrides).length) {
```

Comment:
you could probably do an early return here by moving to top if overrides is empty